File: app/src/main/java/com/bubelov/coins/sync/DatabaseSyncService.java
package com.bubelov.coins.sync;

import android.app.IntentService;
import android.content.Context;
import android.content.Intent;

import com.bubelov.coins.dagger.Injector;

/**
 * @author Igor Bubelov
 */

public class DatabaseSyncService extends IntentService {
    public static void start(Context context) {
        context.startService(new Intent(context, DatabaseSyncService.class));
    }

    public DatabaseSyncService() {
        super(DatabaseSyncService.class.getSimpleName());
    }

    @Override
    protected void onHandleIntent(Intent intent) {
        if (intent == null) {
            return;
        }

        DatabaseSync databaseSync = Injector.INSTANCE.mainComponent().databaseSync();

        if (!databaseSync.isSyncing()) {
            databaseSync.sync();
        }
    }
}

File: app/src/main/java/com/bubelov/coins/database/sync/DatabaseSyncService.kt
package com.bubelov.coins.database.sync

import android.app.IntentService
import android.content.Context
import android.content.Intent

import dagger.android.AndroidInjection
import javax.inject.Inject

/**
 * @author Igor Bubelov
 */

class DatabaseSyncService : IntentService(DatabaseSyncService::class.java.simpleName) {
    @Inject internal lateinit var databaseSync: DatabaseSync

    override fun onCreate() {
        AndroidInjection.inject(this)
        super.onCreate()
    }

    override fun onHandleIntent(intent: Intent?) {
        databaseSync.sync()
    }

    companion object {
        fun start(context: Context) {
            context.startService(Intent(context, DatabaseSyncService::class.java))
        }
    }
}